docs/guides/deployment/fly_io.rst

```diff
@@ -128,7 +127,7 @@ this command:
 
 .. code-block:: bash
 
-    $ flyctl machine update <machine-id> --memory 512 --app $PG_APP -y
+    $ flyctl machine update <machine-id> --memory 1024 --app $PG_APP -y
```

Comment:
I wonder what is going on there.  We have no problem bootstrapping Postgreses on EdgeDB Cloud with much smaller allocations.

Reply:
These are the postgres logs when it only has 512MB.

```
2023-12-22T16:00:06Z app[4d89d19dc012d8] lax [info]admin    | Running...
2023-12-22T16:00:06Z app[4d89d19dc012d8] lax [info]exporter | Running...
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]monitor  | Monitoring dead members
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]monitor  | Monitoring cluster state
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]monitor  | Monitoring replication slots
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]monitor  | Pruning every 24h0m0s...
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:07] [NOTICE] repmgrd (repmgrd 5.3.3) starting up
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:07] [INFO] connecting to database "host=fdaa:0:7c91:a7b:21a:c3cf:7894:2 port=5433 user=repmgr dbname=repmgr connect_timeout=5"
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:07] [ERROR] connection to database failed
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:07] [DETAIL]
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | connection to server at "fdaa:0:7c91:a7b:21a:c3cf:7894:2", port 5433 failed: Connection refused
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  |   Is the server running on that host and accepting TCP/IP connections?
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  |
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:07] [DETAIL] attempted to connect using:
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  |   user=repmgr connect_timeout=5 dbname=repmgr host=fdaa:0:7c91:a7b:21a:c3cf:7894:2 port=5433 fallback_application_name=repmgr options=-csearch_path=
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.164 UTC [328] LOG:  starting PostgreSQL 15.3 (Debian 15.3-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.164 UTC [328] LOG:  listening on IPv4 address "0.0.0.0", port 5433
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.164 UTC [328] LOG:  listening on IPv6 address "::", port 5433
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | exit status 6
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]repmgrd  | restarting in 5s [attempt 1]
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.167 UTC [328] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5433"
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.169 UTC [365] LOG:  database system was shut down at 2023-12-22 16:00:04 UTC
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:00:07.176 UTC [328] LOG:  database system is ready to accept connections
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (330) : New worker (369) forked
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (330) : Loading success.
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (369) : bk_db/pg1 changed its IP from (none) to fdaa:0:7c91:a7b:21a:c3cf:7894:2 by flydns/dns1.
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (369) : Server bk_db/pg1 ('lax.fmoor-test-postgres.internal') is UP/READY (resolves again).
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (369) : Server bk_db/pg1 administratively READY thanks to valid DNS answer.
2023-12-22T16:00:07Z app[4d89d19dc012d8] lax [info]Voting member(s): 1, Active: 1, Inactive: 0, Conflicts: 0
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | Process exited 0
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | restarting in 1s [attempt 1]
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (330) : haproxy version is 2.8.3-1~bpo12+1
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (330) : path to executable is /usr/sbin/haproxy
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | [ALERT]    (330) : Current worker (369) exited with code 143 (Terminated)
2023-12-22T16:00:08Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (330) : All workers exited. Exiting... (0)
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | Running...
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (393) : New worker (396) forked
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | [NOTICE]   (393) : Loading success.
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (396) : bk_db/pg1 changed its IP from (none) to fdaa:0:7c91:a7b:21a:c3cf:7894:2 by flydns/dns1.
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (396) : Server bk_db/pg1 ('lax.fmoor-test-postgres.internal') is UP/READY (resolves again).
2023-12-22T16:00:09Z app[4d89d19dc012d8] lax [info]proxy    | [WARNING]  (396) : Server bk_db/pg1 administratively READY thanks to valid DNS answer.
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | Running...
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:12] [NOTICE] repmgrd (repmgrd 5.3.3) starting up
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:12] [INFO] connecting to database "host=fdaa:0:7c91:a7b:21a:c3cf:7894:2 port=5433 user=repmgr dbname=repmgr connect_timeout=5"
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | INFO:  set_repmgrd_pid(): provided pidfile is /tmp/repmgrd.pid
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:12] [NOTICE] starting monitoring of node "fdaa:0:7c91:a7b:21a:c3cf:7894:2" (ID: 2138927143)
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:12] [INFO] "connection_check_type" set to "ping"
2023-12-22T16:00:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:00:12] [NOTICE] monitoring cluster primary "fdaa:0:7c91:a7b:21a:c3cf:7894:2" (ID: 2138927143)
2023-12-22T16:00:46Z app[4d89d19dc012d8] lax [info]2023/12/22 16:00:46 New SSH Session - fmoor@edgedb.com
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info][  150.759026] Out of memory: Killed process 845 (postgres) total-vm:455336kB, anon-rss:297292kB, file-rss:0kB, shmem-rss:33072kB, UID:999 pgtables:848kB oom_score_adj:0
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.243 UTC [328] LOG:  server process (PID 845) was terminated by signal 9: Killed
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.243 UTC [328] DETAIL:  Failed process was running:
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |           WITH intro(c) AS (SELECT/*<pg.SelectStmt at 0x7fd703baae10>*/
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |           to_json(json_build_object('_tname',
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |           (SELECT/*<pg.SelectStmt at 0x7fd6f1a048d0>*/
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                   to_json("str__tname~2"."name_value~3") AS "_tname_serialized~1"
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |               FROM
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                   LATERAL
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                   (SELECT/*<pg.SelectStmt at 0x7fd6f66ff710>*/
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                           "str_name~2"."name_value~2" AS "name_value~3"
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                       FROM
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                           LATERAL
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                           (SELECT/*<pg.SelectStmt at 0x7fd6f3a07850>*/
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                   "q~2"."name_value~1" AS "name_value~2"
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                               FROM
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                   LATERAL
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                   (SELECT/*<pg.SelectStmt at 0x7fd6f39f1150>*/
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                           "ObjectType~1"."9f6dd92a-0a42-5341-b599-fbf5494fd973" AS "name_value~1"
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                       FROM
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                           LATERAL
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres |                                           (SELEC
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.244 UTC [328] LOG:  terminating any other active server processes
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.249 UTC [328] LOG:  all server processes terminated; reinitializing
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.274 UTC [854] LOG:  database system was interrupted; last known up at 2023-12-22 16:00:07 UTC
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.276 UTC [857] FATAL:  the database system is in recovery mode
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.303 UTC [858] FATAL:  the database system is in recovery mode
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.309 UTC [859] FATAL:  the database system is in recovery mode
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.315 UTC [860] FATAL:  the database system is in recovery mode
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.481 UTC [854] LOG:  database system was not properly shut down; automatic recovery in progress
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.485 UTC [854] LOG:  redo starts at 0/16F23E8
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.489 UTC [862] FATAL:  the database system is not yet accepting connections
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.489 UTC [862] DETAIL:  Consistent recovery state has not been yet reached.
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.676 UTC [864] FATAL:  the database system is not yet accepting connections
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.676 UTC [864] DETAIL:  Consistent recovery state has not been yet reached.
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.791 UTC [854] LOG:  invalid record length at 0/48A2D80: wanted 24, got 0
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.791 UTC [854] LOG:  redo done at 0/48A2348 system usage: CPU: user: 0.15 s, system: 0.12 s, elapsed: 0.30 s
2023-12-22T16:02:37Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:37.797 UTC [855] LOG:  checkpoint starting: end-of-recovery immediate wait
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:38.001 UTC [855] LOG:  checkpoint complete: wrote 6505 buffers (99.6%); 0 WAL file(s) added, 0 removed, 3 recycled; write=0.064 s, sync=0.140 s, total=0.205 s; sync files=3460, longest=0.002 s, average=0.001 s; distance=50882 kB, estimate=50882 kB
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]postgres | 2023-12-22 16:02:38.006 UTC [328] LOG:  database system is ready to accept connections
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | WARNING:  terminating connection because of crash of another server process
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:02:38] [INFO] attempting to reconnect to node "fdaa:0:7c91:a7b:21a:c3cf:7894:2" (ID: 2138927143)
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:02:38] [INFO] reconnected to node "fdaa:0:7c91:a7b:21a:c3cf:7894:2" (ID: 2138927143)
2023-12-22T16:02:38Z app[4d89d19dc012d8] lax [info]repmgrd  | INFO:  set_repmgrd_pid(): provided pidfile is /tmp/repmgrd.pid
2023-12-22T16:02:39Z app[4d89d19dc012d8] lax [info]exporter | ERRO[0152] Error opening connection to database (postgresql://flypgadmin:PASSWORD_REMOVED@[fdaa:0:7c91:a7b:21a:c3cf:7894:2]:5433/postgres?sslmode=disable): driver: bad connection  source="postgres_exporter.go:1658"
2023-12-22T16:02:54Z app[4d89d19dc012d8] lax [info]exporter | ERRO[0167] Error opening connection to database (postgresql://flypgadmin:PASSWORD_REMOVED@[fdaa:0:7c91:a7b:21a:c3cf:7894:2]:5433/repmgr?sslmode=disable): driver: bad connection  source="postgres_exporter.go:1608"
2023-12-22T16:02:55Z app[4d89d19dc012d8] lax [info]exporter | ERRO[0168] Error opening connection to database (postgresql://flypgadmin:PASSWORD_REMOVED@[fdaa:0:7c91:a7b:21a:c3cf:7894:2]:5433/fmoor_test?sslmode=disable): driver: bad connection  source="postgres_exporter.go:1608"
2023-12-22T16:05:07Z app[4d89d19dc012d8] lax [info]monitor  | Voting member(s): 1, Active: 1, Inactive: 0, Conflicts: 0
2023-12-22T16:05:12Z app[4d89d19dc012d8] lax [info]repmgrd  | [2023-12-22 16:05:12] [INFO] monitoring primary node "fdaa:0:7c91:a7b:21a:c3cf:7894:2" (ID: 2138927143) in normal state
```

Reply:
What's the Postgres version there?

Reply:
`psql (15.3 (Debian 15.3-1.pgdg120+1))`

Reply:
```
postgres=# select version();
                                                       version
---------------------------------------------------------------------------------------------------------------------
 PostgreSQL 15.3 (Debian 15.3-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
```